### PR TITLE
Add docs spellcheck and fix all spelling errors

### LINKED ISSUE #5460: Offer: add automated spell-check & markdown linting to docs via GitHub Actions
### Is your feature request related to a problem?

It is related to an ask in the docs that mentions folks can help by spotting typos etc.

### The solution you'd like

I've used this technique for a few repos, including the NUnit dcos, the Marten docs, and my blog. 

Essentially:

* Use GitHub actions for spellcheck & linting the docs.
* In the same PR that introduces the action, remediate the spelling & linting issues that occur. 

Then at the end, you're left with material improvements + the ability to spot issues later.

Let me know if you're up for this, and I'd be happy to adapt what I've already done to benefit your docs too.

### Product

Hot Chocolate

## Progress

* [x] Spell check: #5461
* [x] Enable Markdown linting with all rules disabled #6842
* [ ] Turn on rules and remediate in small batches
  * [ ] MD001
  * [x] MD009 #6843
  * [x] MD012 #6843
  * [ ] MD013
  * [x] MD018 #6843
  * [x] MD022 #6843
  * [ ] MD024
  * [ ] MD025
  * [ ] MD026
  * [ ] MD028
  * [ ] MD029
  * [x] MD031 #6843
  * [x] MD032 #6843
  * [ ] MD033 
  * [x] MD034 #6843
  * [ ] MD036
  * [ ] MD037
  * [ ] MD040
  * [ ] MD041
  * [ ] MD045
  * [ ] MD053
- [ ] Only run Markdown lint when changes are detected. (Last step)

### PR DESCRIPTION
**Pull request by SeanKilleen**
- Adds GitHub Action for cSpell
- Adds docs around how to use it
- Fixes errors found by cSpell to get the build to pass.
- Creates a dictionary of accepted words via `cSpell.json`.

Supports #5460

## Remaining Work

* [x] Separate `words` from `ignoreWords` in cSpell dictionary (words can show up as spellcheck suggestions, ignoreWords are just ignored.)

## ℹ️⚠️ This Workflow will need to be enabled prior to merge

This PR adds a new GitHub Actions workflow. As such, you'll need to choose to allow it to run for the first time.

## ℹ️ I suggest squash-merging this PR

Just a heads up that I made very small commits while working on this, but you probably will want to squash-merge this commit instead of bringing it in as 120 smaller commits.

### CODE REVIEW

**Comment by SeanKilleen** [comment]
ℹ️ This is a work in progress. I'll need one of the ChilliCream folks to allow the workflow to run in this PR so that I can continue to iterate on it.

**Comment by SeanKilleen** [comment]
At the outset, we have 460 errors flagged by cSpell. I'll begin to whittle those down now.

**Comment by SeanKilleen** [comment]
Down to 139 spellcheck errors now.

**Comment by SeanKilleen** [comment]
All spelling errors have now been resolved. 🎉

**Comment by SeanKilleen** [comment]
OK, I am realizing I'm way in over my head on the markdown stuff and it's a bigger project than I'd realized to put it all together.

I am going to split those commits into a separate PR after this spellcheck PR is squashed & merged.

**Comment by SeanKilleen** [comment]
OK, I think this is now ready for the workflow to be tried and the changes to be reviewed. 👍

File: .github/workflows/docs-linting.yml
Comment (SeanKilleen) [inline .github/workflows/docs-linting.yml:12]: ℹ️ Note: this is commented specifically because I'm choosing to split up the Markdown linting PR. If you feel it necessary to delete in this PR, that's fine and I'll do so. 👍

File: CHANGELOG.md
Comment (SeanKilleen) [inline CHANGELOG.md:114]: ℹ️ My auto-save formatting appears to have made some additional minor changes here. I don't think it's an issue but if you need me to revert these specific changes, I will.

File: CONTRIBUTING.md
Comment (SeanKilleen) [inline CONTRIBUTING.md:81]: ℹ️ Similar to the above, I can remove the markdown lint related text if desired prior to merge.

File: cSpell.json
Comment (SeanKilleen) [inline cSpell.json:62]: ℹ️ Technically not correct JSON, but accepted by the tooling and in use in other repositories (I took this from one where I'm currently using it).

**Comment by SeanKilleen** [comment]
@michaelstaib not asking for a review on this -- I know there are other priorities -- but if you could click to enable the workflow so that I can fix it if it doesn't work, I'd really appreciate it. 👍

**Comment by michaelstaib** [comment]
I will have a look later today but already saw that the surfice to surface is wrong. It must be suffice. Will have a closer look later.

Thank you for the help.

File: CONTRIBUTING.md
Comment (michaelstaib) [inline CONTRIBUTING.md:76]: I think these should be included into the website tasks. @rstaib what do you think? should they be added to the docs build steps?

**Comment by michaelstaib** [comment]
@SeanKilleen are you on slack? slack.chillicream.com

**Comment by SeanKilleen** [comment]
> @SeanKilleen are you on slack? slack.chillicream.com

Just joined the org 👍

**Comment by SeanKilleen** [comment]
Hi all -- I'd love to get this merged sooner than later because of the number of files it modifies. Please let me know what changes I can make so that we can get it in and avoid merge conflicts.

File: website/src/blog/2021-07-20-rider-language-injection/2021-07-20-rider-language-injection.md
Comment (tobias-tengler) [inline website/src/blog/2021-07-20-rider-language-injection/2021-07-20-rider-language-injection.md:109]:
```suggestion
Now all `.gql` and `.graphql` files in your project will have proper syntax highlighting, IntelliSense and linting.
```

File: website/src/docs/hotchocolate/api-reference/filtering.md
Comment (tobias-tengler) [inline website/src/docs/hotchocolate/api-reference/filtering.md:882]:
```suggestion
| _You want to configure how _Hot Chocolate_ generates the name and the description of filters in globally? e.g. `PascalCaseFilterType`?_ | Filter&nbsp;Conventions         |
```

File: website/src/docs/hotchocolate/integrations/spatial-data.md
Comment (tobias-tengler) [inline website/src/docs/hotchocolate/integrations/spatial-data.md:597]:
```suggestion
## Re-projection
```

File: website/src/docs/hotchocolate/v10/schema/code-first-object-type.md
Comment (tobias-tengler) [inline website/src/docs/hotchocolate/v10/schema/code-first-object-type.md:205]:
```suggestion
        desc.BindFields(BindingBehavior.Explicit);
```

**Review by tobias-tengler — APPROVED** [review]
I left some small suggestions, but apart from those this seems like a wonderful addition.

Thank you so much for your work on this 🙏

**Comment by SeanKilleen** [comment]
@tobias-tengler great suggestions! Applied them all. This is good to go I think. 👍

**Comment by SeanKilleen** [comment]
Hmm...I see failed builds but AFAICT they're not related to my changes. @tobias-tengler let me know if I'm missing something please, and I'll get it fixed up.

**Comment by tobias-tengler** [comment]
The build failures seem unrelated.
@michaelstaib Are the failing builds expected to be flaky? I can't merge without them passing, but maybe you can!?

**Comment by michaelstaib** [comment]
Yes ... we are working on this ... we will get them green towards mid November ... do not worry.

**Comment by michaelstaib** [comment]
@tobias-tengler you approved it ... shall I merge or do another review if the fixes?

**Comment by tobias-tengler** [comment]
@michaelstaib it should be fine to merge!

**Comment by michaelstaib** [comment]
@SeanKilleen many thanks for your work ... it is not everyday that we get such help!

**Comment by SeanKilleen** [comment]
Thanks all! Happy to contribute to such a great project. You're doing awesome work!

Since this happens to be Hacktoberfest, if you feel so inclined, please add a `hacktoberfest-approved` label to this PR. If not, no worries; that's not why I do it. :)

**Comment by SeanKilleen** [comment]
CC @michaelstaib / @tobias-tengler on the above for your consideration. 👍 If not, no worries.

**Comment by tobias-tengler** [comment]
@michaelstaib I wasn't aware of Hacktoberfest. Seems like a good cause. Maybe we can opt-in our repo and also label our "good first contribution" issues to attract Hacktoberfest contributors: https://hacktoberfest.com/participation/#maintainers

**Comment by SeanKilleen** [comment]
@tobias-tengler I think that'd be an awesome idea, as I do think it's a good cause. In the past, it used to be a bit problematic for maintainers because it produced a flood of low-quality PRs but they've done a _lot_ of tweaking and I haven't seen those issues re-surface in the past years.

**Comment by SeanKilleen** [comment]
@tobias-tengler one more change, I'm sorry! my suggested label of `hacktoberfest-approved` is incorrect. It should be `hacktoberfest-accepted`. Sorry for the mix-up!

**Pull request merged**
